File: leetcode/Python/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        m, sol = defaultdict(int), []
        for n in nums1:
            m[n]=1
        for n in nums2:
            if m[n] > 0:
                m[n]=0
                sol.append(n)
        return sol

File: leetcode/C++/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> visit(1001,0), sol;
        for(int &t : nums1) visit[t]=1;
        for(int &t: nums2) if(--visit[t]==0) sol.push_back(t);
        return sol;
    }
};